### PR TITLE
Feature/153812324/payment gateway

### PR DESCRIPTION
**Pull request by Merdoth**
#### What does this PR do?

- Integrate Paystack as a payment method

 #### Description of Task to be completed?

- Integrate paystack API
- Integrate paystack payment for client users

 #### How should this be manually tested?

- Add product to cart and on checkout, the paystack option will be available.

#### What are the relevant pivotal tracker stories?

- story type: feature
- story id: 153812324
- story title: Payment gateway

 ####Any background context you want to provide?

N/A

Screenshots (if appropriate)
N/A

### CODE REVIEW

File: imports/plugins/included/payments-paystack/client/settings/components/paystackSettingsForm.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          <label className="control-label">
+            <Translation defaultValue="Payment provider supported methods" i18nKey="reaction-payments.paymentSettings.supportedMethodsLabel" />
+          </label>
+          <br />
+
+          <div>
+            <Checkbox label="Authorize" onChange={this.handleCheckBox} name="Authorize" checked={this.state.checkbox.Authorize} />
+          </div>
+
+          {/* <div>
```
Comment (temilaj) [inline imports/plugins/included/payments-paystack/client/settings/components/paystackSettingsForm.js]: Good job working on this PR. It looks good so far. However, can you take out this commented out section? More so, why do we have commented out code in source control?

File: imports/plugins/included/payments-paystack/client/settings/containers/PaystackSettingsFormContainer.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import { TranslationProvider } from "/imports/plugins/core/ui/client/providers";
+import { Reaction, i18next } from "/client/api";
+import { PaystackSettingsForm } from "../components";
+
+class PaystackSettingsFormContainer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      secretKey: "278302390293",
```
Comment (temilaj) [inline imports/plugins/included/payments-paystack/client/settings/containers/PaystackSettingsFormContainer.js]: it's not a good practice to include application settings and secret keys in your codebase. Let's set these as environment variables

File: imports/plugins/included/payments-paystack/server/i18n/ar.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[{
```
Comment (temilaj) [inline imports/plugins/included/payments-paystack/server/i18n/ar.json:1]: DId you add this manually? If so, why?

File: imports/plugins/included/payments-paystack/server/methods/paystack-payment-methods.app-test.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      total: "19.99"
+    };
+
+    const transactionType = "authorize";
+    const transaction = PaystackApi.methods.authorize.call({
+      transactionType: transactionType,
+      cardData: cardData,
+      paymentData: paymentData
+    });
+    expect(transaction).to.not.be.undefined;
```
Comment (temilaj) [inline imports/plugins/included/payments-paystack/server/methods/paystack-payment-methods.app-test.js]: Good job including test suites. However, how can we make this more concrete, as opposed to just testing that  it's defined?

**Review by temilaj — CHANGES_REQUESTED** [review]
Good job with this Pr. I've made a couple comments we'd need to bolster this feature. YOu should also include an email notification feature for all parties involved int he. payment.

File: imports/plugins/included/payments-paystack/server/i18n/ar.json (hunk 3 above)
Comment (Merdoth) [reply imports/plugins/included/payments-paystack/server/i18n/ar.json:1]: No I didn't,  just edited the file to accommodate new changes and updates needed to make the feature explicit.

File: imports/plugins/included/payments-paystack/client/settings/components/paystackSettingsForm.js (hunk 1 above)
Comment (Merdoth) [reply imports/plugins/included/payments-paystack/client/settings/components/paystackSettingsForm.js]: Wrote some codes which weren't working as supposed to and inorder to fix the 
bug I had to debug by creating different instances of the code and  work through each of them one after the order. I guess I forgot to remove this particular piece

File: imports/plugins/included/payments-paystack/server/methods/paystack-payment-methods.app-test.js (hunk 4 above)
Comment (damakahindi) [reply imports/plugins/included/payments-paystack/server/methods/paystack-payment-methods.app-test.js]: Hey @Merdoth could you deal with this. It's a good for a first try. But could you test exactly what should be returned back.

**Review by temilaj — CHANGES_REQUESTED** [review]
Your commit messages have begun to derail from the convention. Please look through these and edit these commits.

**Comment by damakahindi** [comment]
Merging this PR but look into your commit messages.

**Pull request merged**